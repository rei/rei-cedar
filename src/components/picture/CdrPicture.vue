<script setup lang="ts">
import { useCssModule } from 'vue';
import CdrImg from '../image/CdrImg.vue';
import { baseImageProps, pictureSourceObject } from '../../types/interfaces';

/** Media for capturing attention and communicating messages while allowing for art-direction of the image */
defineOptions({
  name: 'CdrPicture',
});

interface pictureProps extends baseImageProps {
  /** An object containing 1..n child objects containing HTMLSourceElement attributes */
  sources: pictureSourceObject
}

withDefaults(defineProps<pictureProps>(), {
  alt: '',
});

const baseClass = 'cdr-picture';
const style = useCssModule();
</script>

<template>
  <picture
    :class="style[baseClass]"
  >
    <source 
      v-for="(source, i) in sources"
      :key="`${source.media}--${i}`"
      v-bind="source"
    >
    <CdrImg
      :src="src"
      :alt="alt"
      :ratio="ratio"
      :fit="fit"
      :position="position"
      :loading="loading"
      :fetchpriority="fetchpriority"
      :decoding="decoding"
    />
  </picture>
</template>

<style lang="scss" module src="./styles/CdrPicture.module.scss">
</style>