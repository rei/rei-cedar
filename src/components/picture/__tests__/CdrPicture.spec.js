import { shallowMount } from '../../../../test/vue-jest-style-workaround.js';
import CdrPicture from '../CdrPicture.vue';

describe('CdrPicture', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrPicture, {
      props: {
        src: '/src/dev/static/cedar-350x150.jpg',
        sources: {
          xs: {
            media: '(max-width: 768px)',
            srcset: '/src/dev/static/cedar-350x150.jpg',
            width: '300',
          },
          sm: {
            media: '(min-width: 768px)',
            srcset: '/src/dev/static/cedar-1920x1080.jpg',
          },
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
