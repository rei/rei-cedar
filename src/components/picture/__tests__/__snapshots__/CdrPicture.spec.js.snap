// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrPicture > matches snapshot 1`] = `
<picture
  class="cdr-picture"
>
  
  <source
    media="(max-width: 768px)"
    srcset="/src/dev/static/cedar-350x150.jpg"
    width="300"
  />
  <source
    media="(min-width: 768px)"
    srcset="/src/dev/static/cedar-1920x1080.jpg"
  />
  
  <cdr-img-stub
    alt=""
  />
</picture>
`;
